### PR TITLE
images: FIXES change image name when updating.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
The current implementation is allowing to change the image name when updating an image via API, this should not be allowed as image names are refs. 

This PR fixes this behavior by:
1. if the update image name is empty, set it to previous image name. 
2. if the update image name is not empty and different from the previous image name return error.

FIXES: THEEDGE-2128

## Type of change

What is it?

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [X] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

**Comment by ldjebran** [comment]
@tomershinhar , this will need automation tests to be updated :smiley:

**Comment by ldjebran** [comment]
Many thanks @tomershinhar for updating the automation tests.

**Review by de1987 — APPROVED** [review]
LGTM!

Just observation, if we have integrations tests instead of test routes and tests separated would improve our confidence, and also avoid any error that mocking might cause.
Unit tests are useful but for specific cases or error conditions.

In my opinion, combing them will add more confidence to push to prod.

**Comment by de1987** [comment]
> @de1987 that's what we have in this particular case , unittests + automation tests

we discussed it and we are good to go!

File: pkg/routes/images.go [hunk 1]
```diff
@@ -344,6 +345,15 @@ func initImageCreateRequest(w http.ResponseWriter, r *http.Request) (*models.Ima
 	if err := readRequestJSONBody(w, r, ctxServices.Log, &image); err != nil {
 		return nil, err
 	}
+
+	if image.Name == "" {
+		// look if a previous image exists, to handle image name properly.
+		if previousImage, ok := r.Context().Value(imageKey).(*models.Image); ok && previousImage != nil {
```
[the remark this answers is not in the document]
Comment (ldjebran) [reply pkg/routes/images.go:351]: If imageKey does not exist in the context => the value of r.Context().Value(imageKey)   will be nil , casting nil with (*models.Image) result to previousImage nil and ok will be false, even if nil was saved in context key ok will be false.
When we are updating image and previousImage is nil, we are not able to reach this line as this will be validated by ImageByIDCtx.

We are able to reach this line of code, with previousImage nil only when creating a new image  with empty name that is covered here https://github.com/RedHatInsights/edge-api/blob/7883cc299578e9132560a74ff3b7cf22d9a62a96/pkg/routes/images_test.go#L58

in any case if a previousImage is nil and image name is empty the code will continue and this will result in error in ValidateRequest later in this code as demonstrated when creating image .

Please let me know if still need to create a test with previousImage with nil value.  The only way todo that is create a unittest only for function initImageCreateRequest but this will be the same scenario as with create image as this function does not make difference we are creating or updating.

**Review by omaciel — APPROVED** [review]
/lgtm
